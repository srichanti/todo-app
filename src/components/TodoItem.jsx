import React from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'

const StyledList = styled.li`
  list-style: none;
  overflow: hidden;
  width: 100%;
  margin-bottom: 10px;
`

const StyledLabel = styled.label`
  float: left;
  cursor: pointer;
`

const StyledButton = styled.button`
  float: right;
  background: palevioletred;
  color: #fff;
  border-radius: 3px;
  border: 2px solid palevioletred;
  padding: 3px 10px;
  outline: none;
  cursor: pointer;
`

const TodoItem = props => {
  const { id, title } = props
  return (
    <StyledList>
      <StyledLabel htmlFor={id}>
        <input type="checkbox" id={id} /> {title}
      </StyledLabel>
      <StyledButton type="button">Delete</StyledButton>
    </StyledList>
  )
}
TodoItem.propTypes = {
  title: PropTypes.string.isRequired,
  id: PropTypes.string.isRequired
}
export default TodoItem
