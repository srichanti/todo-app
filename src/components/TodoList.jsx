import React from 'react'
import styled from 'styled-components'
import TodoItem from './TodoItem'

const StyledUl = styled.ul`
    padding: 0
`

const TodoList = () => (
    <StyledUl>
        <TodoItem id="todo-1" title="Learn React.js" />
        <TodoItem id="todo-2" title="Learn Vue.js" />
        <TodoItem id="todo-3" title="Learn Laravel" />
    </StyledUl>
)
export default TodoList