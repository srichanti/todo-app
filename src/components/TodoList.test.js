import React from 'react'
import { shallow, mount } from 'enzyme'
import TodoList from './TodoList'
import TodoItem from './TodoItem'

describe('TodoList Component', () => {
  let shallowComponent
  let mountComponent
  let mockdata

  beforeEach(() => {
    mockdata = [
      { id: 1, name: 'jest', isComplete: false },
      { id: 2, name: 'mocha', isComplete: false },
      { id: 3, name: 'enzyme', isComplete: false }
    ]
    shallowComponent = shallow(<TodoList todos={mockdata} />)
    mountComponent = mount(<TodoList todos={mockdata} />)
  })
  it('renders correctly', () => {
    expect(shallowComponent).toMatchSnapshot()
  })
  it('should render list with the given strings', () => {
    expect(shallowComponent).toMatchSnapshot()
  })
  it('child components length should be equal to the length of the str', () => {
    expect(mountComponent.find(TodoItem).length).toEqual(mockdata.length)
  })
})
