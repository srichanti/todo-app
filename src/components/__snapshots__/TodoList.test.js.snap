// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoList Component renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <TodoList />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <TodoItem
          id="todo-1"
          title="Learn React.js"
        />,
        <TodoItem
          id="todo-2"
          title="Learn Vue.js"
        />,
        <TodoItem
          id="todo-3"
          title="Learn Lumen"
        />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "id": "todo-1",
          "title": "Learn React.js",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "id": "todo-2",
          "title": "Learn Vue.js",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "id": "todo-3",
          "title": "Learn Lumen",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <TodoItem
            id="todo-1"
            title="Learn React.js"
          />,
          <TodoItem
            id="todo-2"
            title="Learn Vue.js"
          />,
          <TodoItem
            id="todo-3"
            title="Learn Lumen"
          />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "id": "todo-1",
            "title": "Learn React.js",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "id": "todo-2",
            "title": "Learn Vue.js",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "id": "todo-3",
            "title": "Learn Lumen",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
